pkg/gardenlet/controller/federatedseed/extensions/shootstate_control.go

```diff
@@ -83,6 +83,11 @@ func (s *shootStateControl) createShootStateSyncReconcileFunc(ctx context.Contex
 		purpose := extensionObject.GetExtensionSpec().GetExtensionPurpose()
 		state := extensionObject.GetExtensionStatus().GetState()
 
+		if state == nil {
+			s.log.Debugf("Skip Shoot's %s %s extension state with name %s - no state found.", shoot.Name, kind, name)
+			return reconcile.Result{}, nil
+		}
```

Comment:
I am a bit worried about this. As far as I remember I've also added such logic, but then we decided to remove the state from the ShootState if it set to nil in the Extension's state. @plkokanov, wdyt?

Reply:
I guess we could change the logic a bit to work like this:
if the extension state in the shoot state was not nil, but in the current reconciliation it is nil, we should update the ShootState and delete the extension state from it [check this line](https://github.com/gardener/gardener/pull/2331/files/f304afd50eb3d162a025fac78f5f33bc1fd64b1c#diff-5fbd8d379e4dd574afdac838e68337f2R110). Otherwise we do not update if the state in the current reconciliation is nil.

Reply:
Hm, this sounds unrelated to this PR. So probably @danielfoehrKn should revert this change for now and either he or you can open a new one?

Reply:
I would appreciate if @swilen-iwanow or @plkokanov could take over this issue / give it another thought as you are the experts on the topic. 
I ll revert the change.